otel-extensions/src/main/java/org/hypertrace/agent/otel/extensions/config/HypertraceConfig.java

```diff
@@ -143,9 +144,18 @@ static AgentConfig load(String filename) throws IOException {
     if (!builder.hasEndpoint()) {
       builder.setEndpoint(StringValue.newBuilder().setValue(DEFAULT_REPORTING_ENDPOINT).build());
     }
-    if (builder.getTraceReporterType().equals(Config.TraceReporterType.UNSPECIFIED)) {
+    if (!builder.hasMetricEndpoint()) {
+      builder.setMetricEndpoint(
+          StringValue.newBuilder().setValue(DEFAULT_REPORTING_ENDPOINT).build());
```

Comment:
I believe what we want here is to set the value of the metric endpoint to the value of the trace reporting endpoint, rather than having them default to the same value. That way, to get traces and metrics, most users will only have to configure the reporting endpoint. However, if they want to report to a different endpoint (e.g. to export Prometheus metrics) they can override it. I think this should be:
```
if (!builder.hasMetricEndpoint()) {
  builder.setMetricEndpoint(builder.getEndpoint());
}
```
At this point, the builder.getEndpoint is guaranteed to be set (either to the default value or one interpreted from the environment

Reply:
Yeah if we have a single endpoint for both traces and metrics, that would be better. But it will need to be a breaking change. Because as it is now, the endpoint is only for traces, for example, in case of ZIPKIN trace reporter type, the endpoint would be something like `http://localhost:9411/api/v2/spans` which will not work for metrics if the metrics config is not given. 
The best way would be to have a generic endpoint and two other endpoints for traces and metrics if need to override. But since we already are using endpoint for traces, it will be a breaking change.

Reply:
Updated to use reporter endpoint if metrics endpoint is not set